cmd/crank/beta/convert/convert.go

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2024 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package convert contains Crossplane CLI subcommands for migrating Crossplane
+// resources to newer versions or kinds.
+package convert
+
+import (
+	"github.com/crossplane/crossplane/cmd/crank/beta/convert/deploymentruntime"
+	"github.com/crossplane/crossplane/cmd/crank/beta/convert/pipelinecomposition"
+)
+
+// Cmd converts a Crossplane resource to a newer version or a different kind.
+type Cmd struct {
```

Comment:
Maybe a small suggestion for future improvement. Since we know the type we are converting from (composition, controllerconfig, etc), could the command infer it from the kind rather than require user to provide the type? It would simplify the options for conversion and wouldn't require parameter changes when we add more conversion pairs.

Reply:
I thought about something like this. But then got discouraged when I saw that `pipelinecomposition` has an additional flag. So its seems better to have separate sub-commands that can decide on their flags. Having it all under one cmd and infering, then applying flags seemed messy. But not saying its not the way to go in the future if we decide so